tests/runtimes/test_logging_and_parsing.py

```diff
@@ -0,0 +1,126 @@
+        steps=[
+            (
+                "imputer",
+                SimpleImputer(missing_values="", strategy="constant", fill_value="C"),
+            ),
+            ("encoder", OrdinalEncoder()),
+        ]
+    )
+    encoder_to_imputer.fit([["A"], ["B"], ["C"]])
+    return (
+        np.ones((10, 20)),
+        pd.DataFrame(np.zeros((20, 10))),
+        {"a": [1, 2, 3, 4], "b": [5, 6, 7, 8]},
+        [["A"], ["B"], [""]],
+        3,
+        "hello",
+        encoder_to_imputer,
+    )
+
+
+def parse_inputs(my_array, my_df, my_dict: dict, my_list, my_object, my_int, my_str):
+    assert isinstance(my_array, np.ndarray)
+    assert np.all(my_array == np.ones((10, 20)))
+
+    assert isinstance(my_df, mlrun.DataItem)
+    my_df = my_df.as_df()
+    assert my_df.shape == (20, 10)
+    assert my_df.sum().sum() == 0
+
+    assert isinstance(my_dict, dict)
+    assert my_dict == {"a": {0: 1, 1: 2, 2: 3, 3: 4}, "b": {0: 5, 1: 6, 2: 7, 3: 8}}
+
+    assert isinstance(my_list, list)
+    assert my_list == [["A"], ["B"], [""]]
+
+    assert isinstance(my_object, Pipeline)
+    assert my_object.transform(my_list).tolist() == [[0], [1], [2]]
+
+    return [my_str] * my_int
+
+
+def test_parse_inputs_from_mlrun_function():
+    """
+    Run the `parse_inputs_from_mlrun_function` function with MLRun to see the wrapper is parsing the given inputs
+    (`DataItem`s) to the written configuration provided to the wrapper.
+    """
+    # Create the function and run 2 of the previous functions to create a dataset and encoder objects:
+    mlrun_function = mlrun.code_to_function(filename=__file__, kind="job")
+    artifact_path = tempfile.TemporaryDirectory()
+    log_artifacts_and_results_run = mlrun_function.run(
+        handler="log_artifacts_and_results",
+        returns=[
+            "my_array",
+            "my_df:dataset",
+            "my_dict    :dataset",
+            "my_list:  dataset",
+            "my_int",
+            "my_str : result",
+            "my_object: object",
+        ],
+        artifact_path=artifact_path.name,
+        local=True,
+    )
+
+    # Run the function that will parse the data items:
+    parse_inputs_run = mlrun_function.run(
+        handler="parse_inputs",
+        inputs={
+            "my_list:list": log_artifacts_and_results_run.outputs["my_list"],
+            "my_array : numpy.ndarray": log_artifacts_and_results_run.outputs[
+                "my_array"
+            ],
+            "my_df": log_artifacts_and_results_run.outputs["my_df"],
+            "my_object: sklearn.pipeline.Pipeline": log_artifacts_and_results_run.outputs[
+                "my_object"
+            ],
+            "my_dict: dict": log_artifacts_and_results_run.outputs["my_dict"],
+        },
+        returns=["result_list: result"],
+        params={
+            "my_int": log_artifacts_and_results_run.outputs["my_int"],
+            "my_str": log_artifacts_and_results_run.outputs["my_str"],
+        },
```

Comment:
I have to say I don't like the `{"a:b":"c"}` format here. Instead of artificially using an error-prone (and confusing, since it's a string which is a key of a dictionary) colon to separate - use a tuple. For example:
```
{
    "my_list": (log_artifacts_and_results_run.outputs["my_list"], list),
    "my_no_hint_something": "just a value",
    ...
}
```
Then you can parse like you do today, but instead of relying on colons, if the value is a tuple simply take the 1st and 2nd items. WDYT?

Reply:
I thought about it and it has 2 sides: The minus is the tuple input itself, it makes everything kinda annoying and ugly, but the plus is that users can pass the type hint as a type object, and not a string (meaning he can use aliases as well). @yaronha should make the call.